hal/src/nRF52840/spi_hal.cpp

```diff
@@ -448,20 +449,53 @@ void hal_spi_transfer_dma(hal_spi_interface_t spi, void* tx_buffer, void* rx_buf
         return;
     }
 
+    uint8_t* txPtr = nullptr;
+
+    SCOPE_GUARD ({
+        if (!nrfx_is_in_ram(tx_buffer) && txPtr) {
+            free(txPtr);
+        }
+    });
+
     while(spiMap[spi].transmitting) {
         ;
     }
 
     spiMap[spi].dma_user_callback = userCallback;
     if (spiMap[spi].spi_mode == SPI_MODE_MASTER) {
-        SPARK_ASSERT(spiTransfer(spi, (uint8_t *)tx_buffer, (uint8_t *)rx_buffer, length) == length);
+        if (!nrfx_is_in_ram(tx_buffer)) {
+            // Truncate the data to reuse an allocated pool.
+            constexpr uint32_t maxPoolSize = 256;
+            uint32_t poolSize = std::min(length, maxPoolSize);
+            txPtr = (uint8_t*)malloc(poolSize);
+            SPARK_ASSERT(txPtr != nullptr);
```

Comment:
This can potentially be a non-critical error.

Reply:
Okay. Simply return if `nullptr` then.